watchdog/src/lib.rs

```diff
@@ -28,6 +30,9 @@ thread_local! {
     /// The local storage for the data fetched from the external APIs.
     static BLOCK_INFO_DATA: RwLock<HashMap<BitcoinBlockApi, BlockInfo>> = RwLock::new(HashMap::new());
 
+    /// The local storage for the API access.
+    static API_ACCESS: RwLock<ApiAccess> = RwLock::new(ApiAccess::new());
```

Comment:
nit: I guess this applies to other fields here as well. Why do we use a `RwLock`? I think a `RefCell` would be closer to what you need. See https://mmapped.blog/posts/01-effective-rust-canisters.html#use-threadlocal

Reply:
Thank you, done!